chrome/content/zotero/xpcom/dataDirectory.js

```diff
@@ -428,6 +428,24 @@ Zotero.DataDirectory = {
 	},
 	
 	
+	isNewDirOnDifferentDrive: Zotero.Promise.coroutine(function* (oldDir, newDir) {
+		yield this.markForMigration(oldDir, true);
+		let oldMarkerFile = OS.Path.join(oldDir, this.MIGRATION_MARKER);
+		try {
+			// Attempt moving the marker with noCopy
+			yield OS.File.move(oldMarkerFile, OS.Path.join(newDir, this.MIGRATION_MARKER), {noCopy: true});
+		} catch(e) {
+			if (e instanceof OS.File.Error) {
```

Comment:
We should check for the external-device error code (via `unixErrno` and `winLastError`), just so we don't give a totally confusing error message to everyone with incorrect permissions, etc.